init.sls

```diff
@@ -72,20 +72,31 @@ chown_pgdata:
 
 # Deploy users. Sort them, so order is not changing between salt runs
 {% for config in pillar['postgresql']['users']|default({}) %}
-createuser-{{ loop.index }}:
+createuser-{{ config['username'] }}:
   cmd.run:
     - name: psql -t -c "CREATE ROLE {{ config['username'] }} {{ config['options']|default('NOSUPERUSER NOCREATEDB NOCREATEROLE INHERIT LOGIN') }};"
     - unless: psql -t -c "SELECT 1 FROM pg_roles WHERE rolname='{{ config['username'] }}'" |grep -q 1
     - runas: postgres
+{% endfor %}
 
-# The "replication" keyword is not a real database, but just used for permissions in pg_hba.conf
+{% for config in pillar['postgresql']['databases']|default({}) %}
 {% if config['database'] != "replication" %}
-createdb-{{ loop.index }}:
+createdb-{{ config['database'] }}:
   cmd.run:
     - name: psql -t -c "CREATE DATABASE {{ config['database'] }} OWNER {{ config['username'] }};"
```

Comment:
When using a seperate database configuration key, I'd uggest using the `owner` keyword instead of `username` to be more consistent with the SQL keywords.  I'm not entirely sure on this though, as we're currently using `username` instead of `role` when creating a new user as well. So this might actually be more intuitive. 
What do you think?

Reply:
Yes, that is definitely better.